### PR TITLE
Update modular-architecture.md

### PR DESCRIPTION
**Pull request by jkitman**
@dpc @elsirion Based on our call I think this is right?

### CODE REVIEW

**Comment by dpc** [comment]
Can we put description in the boxes itself? It will help to understand if we don't have to reference graph in the list, me thinks.

Also: can we rename add: `fedimintd` and `fedimint-client-lib` (the generic library) and "end application.

Also let's name module crates `<module>-client` and `<module>-server` to signify the names there are of the fedimint module.

**Comment by dpc** [comment]
@jkitman I tired the changes I was thinking about. [Rendered](https://github.com/fedimint/fedimint/blob/abab24611d8f807b33ef308d9bad0c0358aa1e0f/docs/modular-architecture.md). Looks a big ugly, but that's what I had in my mind.

File: docs/modular-architecture.md [hunk 1]
```diff
@@ -3,23 +3,40 @@ Below is a high-level description of the modular architecture of Fedimint.
...
 ```
 
 ## Crate Descriptions
 `core` are the libraries that we make available to module developers
-- `core/api` - module type interfaces (txs, outcomes, consensus items, etc.), database, and API encoding/decoding
-- `core/client` - client interfaces and logic (tx builder, query strategies)
-- `core/server` - server interfaces and logic (federation module, consensus/hbbft code)
+- `core/api` - module type interfaces (txs, inputs, output, etc.), database, and API encoding/decoding
+- `core/client` - client interfaces, types, and logic (tx builder, query strategies)
+- `core/server` - server interfaces, types, and logic (federation module, outcomes, consensus items, consensus/hbbft code)
```
Comment (elsirion) [inline docs/modular-architecture.md:39]: outcomes seem to belong into API

Comment (elsirion) [inline docs/modular-architecture.md:39]: This section seems out of date with the diagram

**Review by elsirion — COMMENTED** [review]
Diagram looks mostly right, only question I have is if we can collapse `fedimint-core` into `fedimint-api`? It's not clear to me why they are separate. Both are dependencies by both client and server, no obvious benefit/cut line for separation.

File: docs/modular-architecture.md (hunk 1 above)
Comment (dpc) [reply docs/modular-architecture.md:39]: Yes, I have modified the graph to move basic descriptions into the graph itself and I didn't update the lists.

**Comment by dpc** [comment]
> Diagram looks mostly right, only question I have is if we can collapse `fedimint-core` into `fedimint-api`? It's not clear to me why they are separate.

Agreed. Which name do you want to keep?

**Comment by elsirion** [comment]
> Which name do you want to keep?

I don't really care, core is less overloaded as a term. But `module-api` should also be renamed into `module-core` in that case.

**Comment by jkitman** [comment]
@dpc I decided to just nuke the text and put everything in the diagram, but cleaned things up a bit.  Let me know if that looks good to you.

**Comment by dpc** [comment]
[It's beautiful](https://github.com/fedimint/fedimint/blob/bc60e49194c58f0dc7a66a35d2e783971e9e4aee/docs/modular-architecture.md)

**Review by dpc — APPROVED** [review]

**Comment by jkitman** [comment]
@okjodom Brought up a good point: where does the gateway and gateway-client fit in?

**Comment by okjodom** [comment]
> @okjodom Brought up a good point: where does the gateway and gateway-client fit in?

My initial impression #651

**Comment by dpc** [comment]
My understanding is that gateway consist of:

* normal client & server module part
* another standalone piece of software that runs along with LN node

The Fedimint module will be like any other module, while the standalone gateway can do whatever it needs. Probably it can be another crate, that either depends on its module crates or shares some crates with it.

**Review by dpc — APPROVED** [review]

**Pull request merged**